### PR TITLE
Added homework of Vladislav Kolev

### PR DESCRIPTION
**Pull request by Vladikolev0321**

### CODE REVIEW

File: homework_03/06_vladislav_kolev/circle.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+from enum import Enum, auto
+from math import sqrt, pow
+
+class RelativePosition(Enum):
```
Comment (petardmnv) [inline homework_03/06_vladislav_kolev/circle.py:4]: OK

File: homework_03/06_vladislav_kolev/circle.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    NO_COMMON_POINTS = 1
+    TOUCHING = 2
+    INTERSECTING = 3
+    SAME = 4
+
+class Point:
```
Comment (petardmnv) [inline homework_03/06_vladislav_kolev/circle.py:10]: OK

File: homework_03/06_vladislav_kolev/circle.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    def __init__(self, x, y):
+        assert isinstance(x, float) and isinstance(y, float)
+        self.x = x
+        self.y = y
+
+
+class Circle:
```
Comment (petardmnv) [inline homework_03/06_vladislav_kolev/circle.py:17]: OK

File: homework_03/06_vladislav_kolev/test_circle.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from circle import Circle, Point, RelativePosition
+
+
+def test_same():
```
Comment (petardmnv) [inline homework_03/06_vladislav_kolev/test_circle.py:4]: OK

File: homework_03/06_vladislav_kolev/test_circle.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    circle1 = Circle(Point(1.0, 1.0), 2.0)
+    circle2 = Circle(Point(1.0, 1.0), 2.0)
+
+    assert circle1.find_relative_position(circle2) == RelativePosition.SAME
+
+def test_intersecting():
```
Comment (petardmnv) [inline homework_03/06_vladislav_kolev/test_circle.py:10]: OK

File: homework_03/06_vladislav_kolev/test_circle.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    circle1 = Circle(Point(0.0, 0.0), 5.0)
+    circle2 = Circle(Point(4.0, 0.0), 6.0)
+    assert circle1.find_relative_position(circle2) == RelativePosition.INTERSECTING
+
+    ## 
+    circle1 = Circle(Point(7.0, 0.0), 4.0)
+    circle2 = Circle(Point(2.0, 0.0), 5.0)
+    assert circle1.find_relative_position(circle2) == RelativePosition.INTERSECTING
+
+def test_touching():
```
Comment (petardmnv) [inline homework_03/06_vladislav_kolev/test_circle.py:20]: OK

File: homework_03/06_vladislav_kolev/test_circle.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    circle1 = Circle(Point(-1.0, 1.0), 3.0)
+    circle2 = Circle(Point(3.0, -2.0), 2.0)
+    assert circle1.find_relative_position(circle2) == RelativePosition.TOUCHING
+
+    # test touching internally
+    circle1 = Circle(Point(-1.0, 0.0), 3.0)
+    circle2 = Circle(Point(3.0, -3.0), 8.0)
+    assert circle1.find_relative_position(circle2) == RelativePosition.TOUCHING
+
+def test_no_common_points():
```
Comment (petardmnv) [inline homework_03/06_vladislav_kolev/test_circle.py:31]: OK

**Review by petardmnv — APPROVED** [review]

File: homework_03/06_vladislav_kolev/circle.py (hunk 2 above)
Comment (StelianRBG) [reply homework_03/06_vladislav_kolev/circle.py:10]: i sugest use other value for enum
0 - no_common_points
1-touching
2-intersecting
3-same

File: homework_03/06_vladislav_kolev/test_circle.py (hunk 6 above)
Comment (StelianRBG) [reply homework_03/06_vladislav_kolev/test_circle.py:20]: you dont need second example

**Review by StelianRBG — COMMENTED** [review]
good decision

**Pull request merged**